libs/datasets/timeseries.py

```diff
@@ -753,6 +753,14 @@ def _diff_preserving_first_value(series):
     return series_diff
 
 
+def _add_new_cases_to_latest(timeseries_df: pd.DataFrame, latest_df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
how about adding `assert latest_df.index.names == [CommonFields.LOCATION_ID]` to catch when we change it and communicate the expectation to people reading the code?